etcdserver/server_test.go

```diff
@@ -869,7 +869,7 @@ func TestSnapshot(t *testing.T) {
 	if !reflect.DeepEqual(gaction[1], testutil.Action{Name: "SaveNoCopy"}) {
 		t.Errorf("action = %s, want SaveNoCopy", gaction[1])
 	}
-	gaction = p.Action()
+	gaction, _ = p.Wait(1)
```

Comment:
This is still racey. p would need to be NewStorageRecorderStream for this to be reliable.